_posts/2024-11-05-how-a-simple-configuration-almost-compromised-our-application.markdown

```diff
@@ -0,0 +1,96 @@
+---
+layout: postv2
+inject_optimize_tag: true
+title: "How a Simple Configuration Almost Compromised Our Application"
+description: "Discover how a single misconfiguration in Amazon CloudFront led to a security vulnerability, exposing user data to potential risk, and the steps we took to resolve it."
+date: 2024-11-05
+highlight: monokai
+author: Caio K. Kaihara
+image: /img/2024-11-05-how-a-simple-configuration-almost-compromised-our-application/hero.png
+tags: [
+  "CloudFront caching",
+  "AWS security",
+  "session hijacking",
+  "Okta integration",
+  "web application security",
+  "AWS Well-Architected Framework"
+]
+---
+
+# How a Simple Configuration Error Almost Compromised Our Application
+
+Imagine logging into an application and suddenly viewing someone else’s account. As alarming as this sounds, it was nearly a reality in one of our applications. In this post, we’ll detail how a simple misconfiguration in Amazon CloudFront led to a security vulnerability and how we managed to resolve it.
+
+#### The Problem: Cross-User Data Exposure
+
+During extended testing, we identified a security vulnerability within the authentication process. Despite using OKTA as our identity provider, we observed sporadic occurrences of session hijacking, where authenticated users unexpectedly accessed other users’ sessions. Although infrequent, this issue posed a significant risk to data security.
+
+Initially, we suspected that the issue was related to user IDs in the database. During development, these IDs were randomly generated with each insertion, meaning a single user could have different IDs over time. This inconsistency, combined with the way we used cookies to store session IDs, seemed to open the door to unauthorised access.
```

Comment:
There's a missing detail here where users need to know you are seeding/generating users per build. I bring this up because the argument is weak, as a ID is static per entry in the database.